frontend/catalyst/jax_primitives.py

```diff
@@ -294,7 +294,9 @@ def _python_callback_lowering(
     ip = module.body
     attrs = [fn_ty_attr, callback_id, len(args), len(results_ty)]
     with ir.InsertionPoint(ip):
-        callbackOp = CallbackOp(f"callback_{callback_id}", *attrs)
+        # TODO: Name mangling for callbacks
```

Comment:
Just out of curiosity, how you imagine this mangling to happen? Isn't enough callback + name + id?

Reply:
I would like to remove the ID part and instead just have `f"{name}{paramtypes}{returntypes}"` or something like that.

Reply:
- The `callback` keyword will be gone?, why keeping it now then?
- How much effort is to do the intended final mangling `f"{name}{paramtypes}{returntypes}"` 
If it's not so hard, I would go for it instead.

Reply:
> How much effort is to do the intended final mangling f"{name}{paramtypes}{returntypes}"
If it's not so hard, I would go for it instead.

It is non trivial. Because parameters are tensors of potentially different types. So we need a map from types to strings and potentially agree on a format for easy readability. I would prefer to leave it for another time as this was not a ticket scheduled for this iteration, just something that facilitated my work on the next PR I will be opening.

Reply:
> The callback keyword will be gone?, why keeping it now then?

If you want it gone, I gave my rationale why I want to keep it, but we can also try removing it.

Reply:
This https://github.com/PennyLaneAI/catalyst/pull/919/files#r1671205731 applies here, I understand. But in your mangling proposal you dropped the `callback` keyword. Maybe you just forgot. I agree, preserving `callback` is beneficial: `f"callback_{name}_{paramtypes}_{returntypes}"`

Reply:
No, do not remove it, I just wanted to understand why you dropped it :)

Reply:
By dropping you mean not doing `f"callback_{callback.__name__}_{callback_id}"`? I just moved callback.__name__ out of the f-string? The line looked too long for my taste :sweat_smile:

Reply:
Thanks for the clarification, we can improve the mangling little by little in future PRs.